modnet/models.py

```diff
@@ -554,79 +566,904 @@ class OR only return the most probable class.
 
         return predictions
 
-    def save(self, filename: str):
-        """Save the `MODNetModel` across 3 files with the same base
-        filename:
-
-        * <filename>.json contains the Keras model JSON dump.
-        * <filename>.pkl contains the `MODNetModel` object, excluding the
-          Keras model.
-        * <filename>.h5 contains the model weights.
+    def evaluate(self, test_data: MODData) -> pd.DataFrame:
+        """Evaluates the target values for the passed MODData by returning the corresponding loss.
 
         Parameters:
-            filename: The base filename to save to.
+            test_data: A featurized and feature-selected `MODData`
+                object containing the descriptors used in training.
+
+
+        Returns:
+            Loss score
+        """
+        # prevents Nan predictions if some features are inf
+        x = test_data.get_featurized_df().replace([np.inf, -np.inf, np.nan], 0)[
+            self.optimal_descriptors[:self.n_feat]
+        ].values
+
+        # Scale the input features:
+        x = np.nan_to_num(x)
+        if self._scaler is not None:
+            x = self._scaler.transform(x)
+            x = np.nan_to_num(x,nan=-1)
+
+        y = []
+        for targ in self.targets_flatten:
+            if self.num_classes[targ] >= 2:  # Classification
+                y_inner = tf.keras.utils.to_categorical(
+                    test_data.df_targets[targ].values,
+                    num_classes=self.num_classes[targ],
+                )
+                loss = "categorical_crossentropy"
+            else:
+                y_inner = test_data.df_targets[targ].values.astype(
+                    np.float, copy=False
+                )
+            y.append(y_inner)
+
+        return self.model.evaluate(x,y)[0]
 
+    def _make_picklable(self):
+        """
+        transforms inner keras model to jsons so that th MODNet object becomes picklable.
         """
 
-        LOG.info("Saving model...")
         model_json = self.model.to_json()
-        with open(f"{filename}.json", "w") as f:
-            f.write(model_json)
-        self.model.save_weights(f"{filename}.h5")
-        try:
-            with open(f"{filename}.history.pkl", "wb") as f:
-                pickle.dump(self.model.history.history, f)
-        except Exception:
-            import traceback
-            traceback.print_exc()
-            LOG.info("Failed to save model history.")
-
-        model = self.model
-        self.model = None
-        self.history = None
-        with open(f"{filename}.pkl", "wb") as f:
-            pickle.dump(self, f)
-        self.model = model
-        LOG.info("Saved model to {}(.json/.h5/.pkl)".format(filename))
+        model_weights = self.model.get_weights()
+
+        self.model = (model_json, model_weights)
```

Comment:
I think there are better ways of doing this; while this approach is fine, the more "Pythonic" way would be to define `__set_state__()` and `__get_state__()`. Have a look at [Pickling Class Instances](https://docs.python.org/3/library/pickle.html#pickle-inst) in the Python docs